### PR TITLE
doc: release notes: add release notes entries for ARM Cortex-M

### PR DESCRIPTION
**Pull request by ioannisg**
Zephyr v2.3 release notes content for ARM Cortex-M architecture.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

### CODE REVIEW

File: doc/releases/release-notes-2.3.rst [hunk 1]
```diff
@@ -132,7 +132,12 @@ Architectures
 
 * ARM:
 
-  * <TBD>
+  * CMSIS is moved out of the main tree and resides in its own standalone module repository
```
Comment (carlescufi) [inline doc/releases/release-notes-2.3.rst:135]:
```suggestion
  * CMSIS has been moved out of the main tree and now resides in its own standalone module repository
```

File: doc/releases/release-notes-2.3.rst [hunk 2]
```diff
@@ -132,7 +132,12 @@ Architectures
...
+  * Cleanups and improvements to the Cortex-M exception vector table
+  * Fix the behavior of Cortex-M spurious IRQ handler
```
Comment (carlescufi) [inline doc/releases/release-notes-2.3.rst:137]:
```suggestion
  * Fixed the behavior of Cortex-M spurious IRQ handler
```

File: doc/releases/release-notes-2.3.rst [hunk 3]
```diff
@@ -132,7 +132,12 @@ Architectures
...
+  * Fix parsing of Cortex-M MemManage Stacking Errors
```
Comment (carlescufi) [inline doc/releases/release-notes-2.3.rst:138]:
```suggestion
  * Fixed parsing of Cortex-M MemManage Stacking Errors
```

File: doc/releases/release-notes-2.3.rst [hunk 4]
```diff
@@ -132,7 +132,12 @@ Architectures
...
+  * Fix arch_cpu_idle() implementation for Cortex-M and Cortex-R
```
Comment (carlescufi) [inline doc/releases/release-notes-2.3.rst:139]:
```suggestion
  * Fixed the arch_cpu_idle() implementation for Cortex-M and Cortex-R
```

**Review by carlescufi — COMMENTED** [review]
Thanks!

**Review by ioannisg — COMMENTED** [review]
@carlescufi will apply yout feedback.

For now I am wondering: should we add a note about Zephyr integration with TF-M as part of Cortex-M release notes or as a standalone entry in the v2.3 release notes? It's not really related to the ARCH porting, but maybe this is the best place to add the entry...

@galak ^^

**Comment by carlescufi** [comment]
> @carlescufi will apply yout feedback.
> 
> For now I am wondering: should we add a note about Zephyr integration with TF-M as part of Cortex-M release notes or as a standalone entry in the v2.3 release notes? It's not really related to the ARCH porting, but maybe this is the best place to add the entry...
> 
> @galak ^^

This will be a headline feature, I will send a PR with those today. But you can also mention it in the arch release notes in more detail.

**Comment by ioannisg** [comment]
@carlescufi feedback applied.
+1 for the added commit by @stephanosio

**Review by carlescufi — APPROVED** [review]

**Comment by ioannisg** [comment]
@stephanosio i think the doc warning is because of line 144 `'cortex_a_r'`, maybe just write cortex_a_r, without extras :)

**Review by MaureenHelm — APPROVED** [review]

**Comment by stephanosio** [comment]
> @stephanosio i think the doc warning is because of line 144 `'cortex_a_r'`, maybe just write cortex_a_r, without extras :)

@ioannisg Indeed, fixed.

**Pull request merged**